Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -539,37 +539,39 @@ checkFields pkg =
-
-  , check (not (null depInternalExecutableWithExtraVersion)) $
-      PackageBuildWarning $
-           "The package has an extraneous version range for a dependency on an "
-        ++ "internal executable: "
-        ++ commaSep (map prettyShow depInternalExecutableWithExtraVersion)
-        ++ ". This version range includes the current package but isn't needed "
-        ++ "as the current package's executable will always be used."
-
-  , check (not (null depInternalExecutableWithImpossibleVersion)) $
-      PackageBuildImpossible $
-           "The package has an impossible version range for a dependency on an "
-        ++ "internal executable: "
-        ++ commaSep (map prettyShow depInternalExecutableWithImpossibleVersion)
-        ++ ". This version range does not include the current package, and must "
-        ++ "be removed as the current package's executable will always be used."
+  -- for more details on why the following was commented out,
+  -- check https://github.com/haskell/cabal/pull/7470#issuecomment-875878507
+  -- , check (not (null depInternalLibraryWithExtraVersion)) $
+  --     PackageBuildWarning $
+  --          "The package has an extraneous version range for a dependency on an "
+  --       ++ "internal library: "
+  --       ++ commaSep (map prettyShow depInternalLibraryWithExtraVersion)
+  --       ++ ". This version range includes the current package but isn't needed "
+  --       ++ "as the current package's library will always be used."
+
+  -- , check (not (null depInternalLibraryWithImpossibleVersion)) $
+  --     PackageBuildImpossible $
+  --          "The package has an impossible version range for a dependency on an "
+  --       ++ "internal library: "
+  --       ++ commaSep (map prettyShow depInternalLibraryWithImpossibleVersion)
+  --       ++ ". This version range does not include the current package, and must "
+  --       ++ "be removed as the current package's library will always be used."
+
+  -- , check (not (null depInternalExecutableWithExtraVersion)) $
+  --     PackageBuildWarning $
+  --          "The package has an extraneous version range for a dependency on an "
+  --       ++ "internal executable: "
+  --       ++ commaSep (map prettyShow depInternalExecutableWithExtraVersion)
+  --       ++ ". This version range includes the current package but isn't needed "
+  --       ++ "as the current package's executable will always be used."
+
+  -- , check (not (null depInternalExecutableWithImpossibleVersion)) $
+  --     PackageBuildImpossible $
+  --          "The package has an impossible version range for a dependency on an "
+  --       ++ "internal executable: "
+  --       ++ commaSep (map prettyShow depInternalExecutableWithImpossibleVersion)
+  --       ++ ". This version range does not include the current package, and must "
+  --       ++ "be removed as the current package's executable will always be used."
```

Comment:
If those are the only ones to be commented, I shall open the issue regarding them, as @phadej suggested.

Reply:
I think no other belong to this spam class, but are you sure all four appear as the spurious warnings?

Reply:
You're actually right, I got ahead of meself, only the first and third ones are to be commented out.